tests/_serialization/test_object_serialization.py

```diff
@@ -61,3 +61,90 @@ def test_bad_object_serde():
 
 def test_object_str_representation():
     assert "<Object[NominalReal] 'foo'>" == str(Object(NominalReal, 'foo'))
+
+
+def test_override_configurations():
+    """Check that weird override cases get caught."""
+    class OverrideTestClass(Serializable['OverrideTestClass']):
+        overridden_value = String("overridden_value", override=True)
+        overridden_option = Optional(String(), "overridden_option", override=True)
+
+    with pytest.raises(AttributeError, match="overridden_value"):
+        OverrideTestClass.build(data={"overridden_value": "string"})
+
+    obj = OverrideTestClass()
+    with pytest.raises(AttributeError, match="overridden_value"):
+        obj.overridden_value = "1"
+    with pytest.raises(AttributeError, match="overridden_option"):
+        obj.overridden_option = "1"
+    with pytest.raises(AttributeError, match="overridden_value"):
+        _ = obj.overridden_value
```

Comment:
What's going on here?

Reply:
Linters complain when a object getter is called in null context.  This tells the linter you know what you are asking for without creating a new symbol.

Reply:
Sorry, I should have been more precise. Why does calling the getting throw an Attribute Error?

Reply:
Sorry, yes, very different question.  In a perfect world, this would be a compile time error because what's getting caught is that you've asked to override a method that doesn't appear in a parent class.  However, when the Property is defined, it doesn't actually know its parentage, and even if it did, it wouldn't know to which class variable you are going to assign it.  The easiest solution then is to throw an error at runtime when you can actually introspect enough to resolve an MRO.